### ISSUE TITLE
Not able to delete a user's public folder after viewing it

### ISSUE THREAD

**Issue by libeclipse** [keybase/client#3323]
Related: #3301 

Opened Chris's folder on first run for testing, now I can't remove it. He also keeps popping up, as mentioned in the related issue, and apparently the only fix for that is to track him.

Deleting /keybase/public/chris does nothing, as it just reappears again.

**Comment by strib** [comment]
Cross-posted from #3301:

@libeclipse it's not really a bug. It sounds like rmdir is successfully removing it, but something on your device (like an indexing program) remembers the path and accesses it again. KBFS can't tell the difference between you intentionally doing that, and some program running as you doing it, so it re-creates the folder when that happens.

Perhaps eventually we'll have a way for you to ignore a folder for all time, and fail all future accesses to it, but I'm not sure it's a good idea to have something so irreversible.

**Comment by libeclipse** [comment]
I agree with you that something so irreversible and permanent is far from a
good solution.

**Comment by inkt-design** [comment]
Is there the possibility of adding a temporary directory "time-stamp" within KBFS? So that it will continually ignore a directory that has recently been removed for a specific period of time, therefore preventing something else on the device (like an indexing program) from re-accessing? Or possibly providing enough time for the offending program to stop, restart or end? 

I could be way out in left field, but that's usually where my thinking comes from.

**Comment by libeclipse** [comment]
Or explicitly add folders to keybasefs instead of just passively detecting if they're accessed. When they're deleted they'd require explicit adding from the client again. 

This would only affect the first time access and adding of a folder, and subsequent behaviour would remain the same.

**Comment by strib** [comment]
We have discussed changing the favoriting model to be more explicit once our GUI is a bit more mature, and I think that's the tentative plan for some time in the future.

I don't think timestamps solve the problem most people see, which is usually with OS "recently accessed" features, and a small delay won't usually help in that case.

**Comment by nehararora** [comment]
Just to provide another data point, this happens to me all the time as well. In my case the culprit is the bird process:
~~~
cat ~/Library/Logs/keybase.kbfs.log | grep 'Pid.*chris'
2017-05-04T09:51:25.404367-04:00 ▶ [DEBU FUSE serve.go:783] e2b <- Lookup [ID=0x2 Node=0x2 Uid=859855370 Gid=20 Pid=67631] "chris"

 ps -ef | grep 67631
859855370 67631     1   0 Wed07AM ??         0:03.39 /System/Library/PrivateFrameworks/CloudDocsDaemon.framework/Versions/A/Support/bird
859855370 68249 64368   0  9:58AM ttys000    0:00.00 grep 67631
~~~
Usually this isn't driven by specific action on my part, but I have seen times when accessing a few different programs causes the folder to be accessed as well.

**Comment by cjb** [comment]
Thanks for tracking the process down, @nehararora!

CC @strib and @gabriel

**Comment by gabriel** [comment]
That bird process is part of iCloud Drive... I'm not sure why it would be remembering old folders. I enabled iCloud Drive on my computer and aren't able to repro.

@nehararora Can you think of any specific steps to reproduce this using iCloud Drive?

**Comment by nehararora** [comment]
@gabriel unfortunately no :(  been seeing this ever since I installed keybase months ago - don't recall doing anything specific. I'll keep a eye on when it pops up again to confirm if I see this from the same process again and whether I can catch anything in the system logs. 

Was a little surprised to see the icloud process accessing the folder myself. Looking at the icloud settings on my mac, don't see anything that (obviously) points at this directory. :|

**Comment by gabriel** [comment]
ok thanks.. yeah only thing I could think of is maybe there is a symlink somewhere that it's following into /keybase
